src/Network/Network/PrivateLinkService/PrivateEndpointConnection/ApproveAzurePrivateEndpointConnection.cs

```diff
@@ -29,46 +29,49 @@ namespace Microsoft.Azure.Commands.Network
     [Cmdlet("Approve", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "PrivateEndpointConnection", DefaultParameterSetName = "ByResourceId"), OutputType(typeof(PSPrivateEndpointConnection))]
     public class ApproveAzurePrivateEndpointConnection : PrivateEndpointConnectionBaseCmdlet
     {
+        [Alias("ResourceName")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource name.",
+            ParameterSetName = "ByResource")]
+        [ValidateNotNullOrEmpty]
+        public override string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The reason of action.")]
+        public string Description { get; set; }
+
         public override void Execute()
         {
             base.Execute();
 
-            string resourceType = string.Empty;
-            string parentResource = string.Empty;
-
             if (this.IsParameterBound(c => c.ResourceId))
             {
                 var resourceIdentifier = new ResourceIdentifier(this.ResourceId);
                 this.ResourceGroupName = resourceIdentifier.ResourceGroupName;
-                this.Name = resourceIdentifier.ResourceName;
-                resourceType = resourceIdentifier.ResourceType;
-                parentResource = resourceIdentifier.ParentResource;
-                this.ServiceName = parentResource.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries).Last();
-            }
-
-            var psPrivateLinkService = this.GetPrivateLinkService(ResourceGroupName, ServiceName);
-            var peConnection = psPrivateLinkService.PrivateEndpointConnections.FirstOrDefault(
-                    resource =>
-                        string.Equals(resource.Name, this.Name, StringComparison.CurrentCultureIgnoreCase));
 
-            if (peConnection == null)
-            {
-                throw new ArgumentException(string.Format(Properties.Resources.ResourceNotFound, this.Name));
+                if (this.ResourceId.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries).Contains("privateEndpointConnections"))
```

Comment:
You case check `resourceIdentifier.ResourceType` instead of splitting id string (applies to other similar code)

Reply:
Thanks for advice. I have realized the way for resourceIdentifier. The main controversy here is the parameter resourceId whether be resource id for private endpoint connection or for its parent resource (Sql/CosmosDB/PrivateLinkService). That's why previously I manually check the resource id to include the private endpoint connection there.
And from general rule for Powershell, the resource id should only be the resource that refer to the current cmdlet. So our PM and I reach an agreement for resource id still referring to private endpoint connection resource id.
So i removed the redundant code you are mentioned about. Thanks.